File: Gestion A C/models/Approvisionnement.php
<?php
require_once 'Operation.php';
require_once 'Fournisseur.php';
require_once 'ArticleConfection.php';
require_once 'Sys.php';

class Approvisionnement extends Operation {
    private static int $nbAp = 0;
    private Fournisseur $fournisseur;
    private string $fournisseur_id;

    public function __construct(Fournisseur $fournisseur = null) {
        parent::__construct();
        self::$nbAp++;
        $this->article = new ArticleConfection();
        $this->id = "AP-" . self::$nbAp;
        $this->nature = "Approvisionnement";
    }

    public function getFournisseur(): Fournisseur {
        return $this->fournisseur;
    }

    public function getFournisseurId(): string {
        return $this->fournisseur_id;
    }

    public function setFournisseur(Fournisseur $fournisseur): void {
        $this->fournisseur = $fournisseur;
        $this->fournisseur_id = $fournisseur->getId();
    }
    
    public function __toString(): string {
        return "\n---------------------------------------\n" .
            "ID                       : " . $this->id . "\n" .
            "ID du fournisseur        : " . $this->fournisseur->getId() . "\n" .
            "Telephone du fournisseur : " . $this->fournisseur->getTelephone() . "\n" .
            "Nom Article              : " . Sys::capitalize($this->article->getNomArticle()) . "\n" .
            "Prix                     : " . $this->article->getPrix() . " Fcfa\n" .
            "Quantite                 : " . $this->article->getQuantite() . "\n" .
            "Montant                  : " . $this->article->getMontant() . " Fcfa\n" .
            "Nature                   : " . $this->nature . "\n" .
            "Date                     : " . $this->date;
    }
}

File: Gestion Atelier De Couture/src/entity/Approvisionnement.java
package entity;

import services.Sys;

public class Approvisionnement extends Operation {
    
    private  static int nbAp;
    private Fournisseur fournisseur;

    public Approvisionnement() {
        super();
        nbAp++;
        this.article = new ArticleConfection();  
        this.id = "AP-" + nbAp;
        this.nature = "Approvisionnement";
    }

    public Fournisseur getFournisseur() {
        return fournisseur;
    }

    public void setFournisseur(Fournisseur fournisseur) {
        this.fournisseur = fournisseur;
    }

    public String toString() {
        return "\n---------------------------------------\n" +
                "ID                       : " + id + "\n" +
                "ID du fournisseur        : " + fournisseur.getId() + "\n" +
                "Telephone du fournisseur : " + fournisseur.getTelephone() + "\n" +
                "Nom Article              : " + Sys.capitalize(article.getNomArticle()) + "\n" +
                "Prix                     : " + article.getPrix() + " Fcfa" + "\n" +
                "Quantite                 : " + article.getQuantite() + "\n" +
                "Montant                  : " + article.getMontant() + " Fcfa" + "\n" +
                "Nature                   : " + nature + "\n"+
                "Date                     : " + date;
    }
}
